src/boot/efi/boot.c

```diff
@@ -2293,18 +2293,16 @@ static EFI_STATUS image_start(
 }
 
 static EFI_STATUS reboot_into_firmware(VOID) {
-        _cleanup_freepool_ CHAR8 *b = NULL;
-        UINTN size;
-        UINT64 osind;
+        UINT64 old, new;
         EFI_STATUS err;
 
-        osind = EFI_OS_INDICATIONS_BOOT_TO_FW_UI;
+        new = EFI_OS_INDICATIONS_BOOT_TO_FW_UI;
 
-        err = efivar_get_raw(GLOBAL_GUID, L"OsIndications", &b, &size);
+        err = efivar_get_uint64(GLOBAL_GUID, L"OsIndications", &old);
         if (!EFI_ERROR(err))
-                osind |= (UINT64)*b;
+                new |= old;
 
-        err = efivar_set_raw(GLOBAL_GUID, L"OsIndications", &osind, sizeof(UINT64), TRUE);
+        err = efivar_set_int(GLOBAL_GUID, L"OsIndications", new, TRUE);
```

Comment:
`efivar_set_uint64()` ?

Reply:
Didn't exist yet but I added it and used it here and below.